### PR TITLE
feat: inject fs size label to pods

### PR DESCRIPTION
**Pull request by TarasPriadka**

### CODE REVIEW

File: modules/nextflow/src/main/groovy/nextflow/Session.groovy [hunk 1]
```diff
@@ -374,6 +379,16 @@ class Session implements ISession {
         // -- file porter config
         this.filePorter = new FilePorter(this)
 
+        // -- fsSize used when provisioning OFS resources
+        if (config.fsSizeTb) {
+            try {
+                this.fsSize = Integer.parseInt(config.fsSizeTb.toString())
```
Comment (AidanAbd) [inline modules/nextflow/src/main/groovy/nextflow/Session.groovy:385]: the fuck is this language? why u go to string? what was it before?

File: modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy [hunk 2]
```diff
@@ -137,6 +139,11 @@ class PodSpecBuilder {
         return this
     }
 
+    PodSpecBuilder withFsSize(int size) {
+        this.fsSize = size
+        return this
+    }
```
Comment (AidanAbd) [inline modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy:145]: this is incredible

**Review by AidanAbd — APPROVED** [review]

File: modules/nextflow/src/main/groovy/nextflow/Session.groovy (hunk 1 above)
Comment (TarasPriadka) [reply modules/nextflow/src/main/groovy/nextflow/Session.groovy:385]: its a generic `Object` type

File: modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy [hunk 3]
```diff
@@ -446,6 +453,10 @@ class PodSpecBuilder {
         if( affinity )
             spec.affinity = affinity
 
+        if (fsSize) {
```
Comment (rahuldesai1) [inline modules/nextflow/src/main/groovy/nextflow/k8s/model/PodSpecBuilder.groovy:456]: prob better to set this to -1 and then check if fsSize == -1

File: modules/nextflow/src/main/groovy/nextflow/Session.groovy [hunk 4]
```diff
@@ -126,6 +126,11 @@ class Session implements ISession {
      */
     Path workDir
 
+    /**
+     * The size of the filesystem to use for the work directory
+     */
+    int fsSize
```
Comment (rahuldesai1) [inline modules/nextflow/src/main/groovy/nextflow/Session.groovy:132]: why does this need to be passed in through the pod spec as a label? 

does it makes sense to just pass this in as a variable when you provision the storage and then store that in the DB, this feels weird no?

**Comment by TarasPriadka** [comment]
No need to change Nextflow, will just pass expected size when creating the OFS mount

**Pull request closed without merging**